### ISSUE TITLE
UI - treemap charts

### ISSUE THREAD

**Issue by vio** [relative-ci/bundle-stats#623]
Rel: #624 
Ref: https://github.com/webpack-contrib/webpack-bundle-analyzer/issues/187

**Comment by jflayhart** [comment]
I think this should be rather trivial since we're already generating the stats files, simply generate the HTML file along with it and link to it

**Comment by jflayhart** [comment]
Which webpack file(s) need to be updated to add webpack-bundle-analyzer and then where would we need to generate the html file to the host server so we can generate a link on the corresponding relative-ci page?

**Comment by vio** [comment]
@jflayhart the webpack-bundle-analyzer is a great tool and I believe they are doing their own parsing of the sources. Will be really complicated if not impossible to pack it in bundle-stats.

[bundle-stats is already extracting and normalising the modules data](https://github.com/relative-ci/bundle-stats/blob/master/packages/utils/src/webpack/extract/__tests__/modules.js#L15) from the webpack stats, so this issue is more about taking that data and displaying it in a tree map chart. The user can switch between the table and the chart view. It will display only the _stat_ size, but from my experience that's a good enough metric to identify the biggest parts of a chunk.

The tree map library has been the main blocker for me, since the idea was to have the bundle-stats.html as light as possible (all the assets are inlined). The ones depending on D3 are great but they can be [quite](https://nivo.rocks/treemap) [large](https://recharts.org/en-US/api/Treemap). I think there are 2 options to unblock this:
1. load the charting lib from a cdn 
2. look for lighter treemap option - for example [bundlephobia is using something custom](https://github.com/pastelsky/bundlephobia/tree/bundlephobia/client/components/Treemap)

**Comment by vio** [comment]
Here is a quick wireframe of the way i see this feature - https://excalidraw.com/#json=6277084896821248,1i-vE6IxVrVGQH8bt8Et-A. Let me know what do you think ;)

**Comment by jflayhart** [comment]
>  the webpack-bundle-analyzer is a great tool and I believe they are doing their own parsing of the sources. Will be really complicated if not impossible to pack it in bundle-stats.

I dont know how your tool works per se, but all these analyzer tools need one thing: the webpack stats.json output file. You should already be generating that with your tool but webpack-bundle-analyser could have an orthogonal relationship with what YOUR tool spits out and simply be used to generate the separate module tree map. The rest is automatically generated into an HTML file that you host and link to somewhere. Which part are you saying is not possible?

I like the dedicated modules tab that's good.

Also, I get you want a lighter lib so that's fair, but webpack-bundle-analyzer is the OFFICIAL one for webpack and should use all the same inputs you should already be generating as the output. Have you tried this and found that's not the case?

**Comment by jflayhart** [comment]
P.S. I guess what I am saying for this UI bit you don't to necessarily pack WBA into bundle-stats for consumers like me, you just want it as a means to an end to generate the module graph on RelativeCI, so bundle-stats shouldn't be affected by this.

**Comment by vio** [comment]
@jflayhart both of the packages are working in a similary way:
1. extract data (webpack-plugin, stats.json file)
2. manipulate 
3. report

What i was trying to convey is that bundle-stats is already having the modules data available, _it only needs_ the chart visualisation. We might not need to bring another full reporting tool just for that(not very knowledgeable of the visualisation lib they are using, at a quick glance is looking like it is not open source).

Here is a high level diagram on how everything is working together:

![bundle-stats-chart](https://user-images.githubusercontent.com/13300/106383595-40b24a00-63c7-11eb-84f1-4dcbe2e44d20.png)

**Comment by vio** [comment]
Available for testing and for feedback on `v4.13.0-beta.2`: 

https://github.com/relative-ci/bundle-stats/assets/13300/5ca9726e-8762-4aa6-9fb4-49c26b2e2226

Related tasks:
- https://github.com/relative-ci/bundle-stats/issues/4316
- https://github.com/relative-ci/bundle-stats/issues/4317

**Comment by vio** [comment]
`bundle-stats@4.13.0-rc.0` 
- grouped assets/modules/packages by folder

https://github.com/relative-ci/bundle-stats/assets/13300/278218ac-3ba8-4cc7-a7ed-4c7501b2ee17

**Comment by vio** [comment]
Released on [`v4.13.0`](https://github.com/relative-ci/bundle-stats/releases/tag/v4.13.0). Feel free to open a new issue if you encounter any problems or have any feedback to share.

https://github.com/relative-ci/bundle-stats/assets/13300/9ab0d11e-4412-4a79-a2b5-1126d32cd9f9
